device/tt_silicon_driver.cpp

```diff
@@ -3063,8 +3059,21 @@ void *tt_SiliconDevice::channel_0_address(std::uint32_t offset, std::uint32_t de
     log_assert(ndesc->is_chip_mmio_capable(device_id), "Cannot call channel_0_address for non-MMIO device");
     struct PCIdevice* pci_device = get_pci_device(device_id);
     auto architecture_implementation = pci_device->hdev->get_architecture_implementation();
-    std::uint64_t bar0_offset = offset - architecture_implementation->get_dram_channel_0_peer2peer_region_start()
-                                + architecture_implementation->get_dynamic_tlb_16m_base() + architecture_implementation->get_dynamic_tlb_16m_size();
+    std::uint64_t bar0_offset;
+
+    // Temporary hack for blackhole bringup.
+    if (arch_name == tt::ARCH::BLACKHOLE) {
+
+        // Use 195th 2MB TLB onwards.
+        const std::uint32_t TLB_CH0_START = 195;
```

Comment:
is there a specific reason that the 195th 2MB TLB is the start?

Reply:
Added a comment, other TLBs before are used as either static of fallback